### ISSUE TITLE
Site Editor: site icon not displayed when set

### ISSUE THREAD

**Issue by vindl** [Automattic/wp-calypso#51482]
If you set the site icon in core it will replace the default WordPress button in the browsing sidebar toggle. The same thing should happen in WP.com integration but it's not working as expected currently.

**Note 1:** It's not possible to set site the icon in core through UI with FSE is enabled at the moment, since we hide the Customizer menu item. You can still reach it directly by appending `wp-admin/customize.php` to your site's URL. 

<img width="304" alt="Screenshot 2021-03-29 at 09 01 36" src="https://user-images.githubusercontent.com/1182160/112798590-64170f80-906d-11eb-9dbe-243db312d470.png">

**Note 2:** It's possible to do this on WP.com through Calypso's settings:

<img width="753" alt="Screenshot 2021-03-29 at 09 26 29" src="https://user-images.githubusercontent.com/1182160/112801171-dccb9b00-9070-11eb-9f5a-12d5f3beb3cc.png">

**Comment by david-szabo97** [comment]
Here are my findings:

Site logo and site title are fetched from the `index` endpoint. In the case of a typical Gutenberg development environment that is `http://localhost:8888/index.php?rest_route=/`.
![image](https://user-images.githubusercontent.com/2256104/113298552-41e6f100-92fc-11eb-86a7-8eb949fdc371.png)


On WP.com this endpoint should live at `https://public-api.wordpress.com/wp/v2/sites/_SITE_ID_/`. But this endpoint is totally different from the WordPress `index` endpoint.
![image](https://user-images.githubusercontent.com/2256104/113298638-51fed080-92fc-11eb-9053-3da42f4e71e7.png)

Related entity is defined here
https://github.com/WordPress/gutenberg/blob/3a9dc1acc4be31f81176cacb01849b9788576f05/packages/core-data/src/entities.js#L21-L26

It uses an empty string as `baseUrl` which means it will call the `index` endpoint.

Any idea how we could access the `index` endpoint in this case?

**Comment by david-szabo97** [comment]
A quick hack that leverages that the endpoint is preloaded: D59634-code

In this case it won't fire a request but load the preloaded endpoint.
https://github.com/WordPress/gutenberg/blob/14f54442424f1490c74b983bc17c7b9c4968287c/lib/full-site-editing/edit-site-page.php#L112

**Comment by david-szabo97** [comment]
Looking forward to any thoughts @mattwiebe @Addison-Stavlo @jeyip

**Comment by vindl** [comment]
cc @creativecoder since he is working on a similar issue in https://github.com/Automattic/wp-calypso/issues/51396.

**Comment by creativecoder** [comment]
To solve this issue in the short term for WP.com, I think something like D59634-code is what we should go with for now. It addresses the troubles I was having resolving Simple vs. Atomic API calls with #51477, so that's helpful!

Long term, I'm not convinced that the site icon url belongs at the root endpoint. It seems like it might belong in settings? But there's a lot of issues to work out--for example, the settings endpoint is only accessible by admin users for the site, currently, and anyone who can load the editor should be able to fetch the site icon url.

**Comment by creativecoder** [comment]
Just found this [trac ticket](https://core.trac.wordpress.org/ticket/50606) and [this comment](https://github.com/WordPress/gutenberg/pull/22952/files#r591129982) as well, further confirming the "hacky-ness" of putting the site icon in the root endpoint and depending on `context=edit` for it to load.

It seems like we should start working on a long term solution for this sooner rather than later.

**Comment by creativecoder** [comment]
I dug into this a little, and here's a Gutenberg PR that removes the need for transforming `/` into `/?context=edit`, which should fix this issue with the site icon: https://github.com/WordPress/gutenberg/pull/30482

**Comment by david-szabo97** [comment]
> I dug into this a little, and here's a Gutenberg PR that removes the need for transforming `/` into `/?context=edit`, which should fix this issue with the site icon: [WordPress/gutenberg#30482](https://github.com/WordPress/gutenberg/pull/30482)

Are you sure that's going to fix it? 🤔 On WP.com it's a proxied request issue and it is not related to the context query parameter. The `/` endpoint is not accessible at all on WP.com which causes this issue. We should move these settings to a totally separate endpoint. (Just like `settings` endpoint)

**Comment by creativecoder** [comment]
> Are you sure that's going to fix it? 🤔 On WP.com it's a proxied request issue and it is not related to the context query parameter.

@david-szabo97 You're right, that Gutenberg change won't fix it (even though I had convinced myself it would 🤦‍♂️)

However, I think I did find a solution! After an epic xdebug session with the wpcom API, I discovered that the root endpoint is available for Simple sites, in the form of `https://public-api.wordpress.com/wp-json/?rest_route=/sites/{site_id}/`. Figuring that out, I put together a diff that gets the site icon url where we need it for Simple sites so the icon can be displayed in the editor.

D59822-code should resolve this issue.
